### PR TITLE
Fixes to cow for CWP release

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes a bug in project creation and allows non-stable releases to include RC tagged dependencies.

### CODE REVIEW

**Comment by dhensby** [comment]
So what happens if there are no tags we can use for a module? We just can't release?

**Comment by tractorcow** [comment]
If a recipe has a dependency marked as upgrade-only and there are no tags, then no you can't release; You can either adjust your recipe dependencies to a lower version, or you can tag a new dependency version.

**Comment by tractorcow** [comment]
@dhensby added a fix to a critical issue.

**Comment by dhensby** [comment]
code style failures! uh-oh

**Comment by tractorcow** [comment]
Fixed.

File: src/Model/Modules/Library.php [hunk 1]
```diff
@@ -338,10 +339,30 @@ public function rebase(OutputInterface $output = null, $remote = 'origin')
...
+        // Set CWD to work-around git crashing while stashing
+        $dir = getcwd();
+        chdir($this->getDirectory());
+
+        // Stash changes
+        $status = $repo->run('status');
+        $hasChanges = stripos($status, 'Changes to be committed:')
+            || stripos($status, 'Changes not staged for commit:');
+        if ($hasChanges) {
+            $repo->run('stash');
```
Comment (dhensby) [inline src/Model/Modules/Library.php]: you might want `git stash -u` which will stash untracked files too...

File: src/Model/Modules/Library.php [hunk 2]
```diff
@@ -839,9 +871,13 @@ protected function createChildLibrary($path)
     public function loadCachedPlan()
     {
         // Check cached plan file
-        $path = $this->getDirectory() . '/.cow.plan.json';
+        $path = $this->getDirectory() . '/.cow.pat.json';
         if (!file_exists($path)) {
-            return null;
+            // Check old name
+            $path = $this->getDirectory() . '/.cow.plan.json';
```
Comment (dhensby) [inline src/Model/Modules/Library.php]: `cow.plan.json` or `cow.pat.json` - you changed it above but not here...

Comment (tractorcow) [inline src/Model/Modules/Library.php]: Yes, this is the legacy fallback. See the `// Check old name` comment.

File: src/Model/Modules/Library.php (hunk 1 above)
Comment (tractorcow) [inline src/Model/Modules/Library.php]: oh I see. Good idea.

**Comment by tractorcow** [comment]
Updated

File: src/Model/Modules/Library.php (hunk 2 above)
Comment (dhensby) [inline src/Model/Modules/Library.php]: Ah, silly me

**Comment by dhensby** [comment]
Merge time or do you want to squash?

**Comment by tractorcow** [comment]
I've squashed into change-per-commit, so that can be squashed on test pass. :)

**Pull request merged**